Modules/Core/SpatialObjects/include/itkSpatialObjectToImageFilter.hxx

```diff
@@ -312,7 +311,27 @@ SpatialObjectToImageFilter<TInputSpatialObject, TOutputImage>::GenerateData()
   {
     region.SetSize(size);
   }
-  region.SetIndex(index);
+
+  IndexType index;
+  index.Fill(0);
+  specified = false;
+  for (i = 0; i < OutputImageDimension; ++i)
+  {
+    if (m_Index[i] != 0)
```

Comment:
This logic seems entirely unnecessary. If `SetIndex` has not been invoked, `m_Index` will be all zeroes because it was default initialized to that, and we don't need an additional `index` variable of all zeroes. If `m_Index` was set, we want to use it. So we can directly use `m_Index` in either case. So the entire set of changes for lines 314 to 334 should be replaced by a single line: `region.SetIndex(m_Index);`.

Reply:
:) - you're right!   Got carried away with cut-n-paste.   Will fix.